newIDE/app/src/Profile/Subscription/SubscriptionDialog.js

```diff
@@ -135,23 +139,23 @@ export default function SubscriptionDialog({
 
   const buyUpdateOrCancelPlan = async (
     i18n: I18nType,
-    plan: { planId: null | string }
+    subscriptionPlanPrice: SubscriptionPlanPrice | null
   ) => {
     const { getAuthorizationHeader, subscription, profile } = authenticatedUser;
     if (!profile || !subscription) return;
-    sendChoosePlanClicked(plan.planId);
-
+    if (subscriptionPlanPrice) {
+      sendChoosePlanClicked(subscriptionPlanPrice.planId);
+    }
     // Subscribing from an account without a subscription
-    if (!subscription.planId) {
+    if (!subscription.planId && subscriptionPlanPrice) {
       setSubscriptionPendingDialogOpen(true);
-      const isEducationPlan = plan.planId === 'gdevelop_education';
-      const planId = isEducationPlan
-        ? `gdevelop_education_${educationPlanPeriodicity}`
-        : plan.planId || '';
+      const isEducationPlan =
+        subscriptionPlanPrice &&
+        subscriptionPlanPrice.planId === 'gdevelop_education';
       const quantity = isEducationPlan ? educationPlanSeatsCount : undefined;
       Window.openExternalURL(
-        getRedirectToCheckoutUrl({
-          planId,
+        getRedirectToCheckoutUrlV2({
+          pricingSystemId: subscriptionPlanPrice.id,
```

Comment:
I find `pricingSystem` quite a convoluted name. Should we rename it to `planPrice` everywhere?